File: LeetCode/C++/53_Maximum_Subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int best = INT_MIN;
        int curr = 0;
        for(int i = 0; i<size(nums);i++) {
            curr=max(nums[i]+curr, nums[i]);
            best=max(best, curr);
        }
        return best;
    }
};

File: LeetCode/Python/53_Maximum_Subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        current_best=-10**6
        total_best=-10**6
        for i in nums:
            current_best = max(i, current_best+i)
            total_best = max(current_best, total_best)
        return total_best
    